r/R/expression.R

```diff
@@ -10,9 +10,21 @@
 #
 #' @export
 runExpression <- function(req, data) {
-  # Get the annotation matrix with the geneid to name translation, and the subset with the correct names.
-  df <- data@misc$gene_annotations
-  genesSubset <- subset(df, toupper(df$name) %in% toupper(req$body$genes))
-  genesSubset <- genesSubset[, c("input", "name")]
-  return(getExpressionValues(genesSubset, data))
+    #Get the annotation matrix with the geneid to name translation, and the subset with the correct names.
+    df <- data@misc$gene_annotations
+    genesSubset <- subset(df, toupper(df$name) %in% toupper(req$body$genes))
+
+    if (!nrow(genesSubset)) {
+
+        stop(
+            generateErrorMessage(
+                "R_WORKER_GENE_NOT_FOUND",
```

Comment:
would be nice to have the error codes in a single place (not a must for this PR)

Reply:
refactored